cmd/kubectl-gadget/top/block-io.go

```diff
@@ -15,43 +15,43 @@
 package top
 
 import (
-	"fmt"
-
 	"github.com/spf13/cobra"
 
+	commontop "github.com/inspektor-gadget/inspektor-gadget/cmd/common/top"
 	commonutils "github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
-	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/top"
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/kubectl-gadget/utils"
 	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/top/block-io/types"
 )
 
 func newBlockIOCmd() *cobra.Command {
-	var commonTopFlags CommonTopFlags
+	var commonFlags utils.CommonFlags
+	var flags commontop.CommonTopFlags
 
 	cols := types.GetColumns()
 
-	cmd := &cobra.Command{
-		Use:   fmt.Sprintf("block-io [interval=%d]", top.IntervalDefault),
-		Short: "Periodically report block device I/O activity",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			parser, err := commonutils.NewGadgetParserWithK8sInfo(&commonTopFlags.OutputConfig, types.GetColumns())
-			if err != nil {
-				return commonutils.WrapInErrParserCreate(err)
-			}
-
-			gadget := &TopGadget[types.Stats]{
-				name:           "biotop",
-				commonTopFlags: &commonTopFlags,
-				parser:         parser,
-				nodeStats:      make(map[string][]*types.Stats),
-				colMap:         cols.GetColumnMap(),
-			}
-
-			return gadget.Run(args)
-		},
-		Args: cobra.MaximumNArgs(1),
-	}
-
-	addCommonTopFlags(cmd, &commonTopFlags, &commonTopFlags.CommonFlags, cols.ColumnMap, types.SortByDefault)
+	cmd := commontop.NewBlockIOCmd(func(cmd *cobra.Command, args []string) error {
+		parser, err := commontop.NewBlockIOParserWithK8sInfo(&commonFlags.OutputConfig, &flags)
+		if err != nil {
+			return commonutils.WrapInErrParserCreate(err)
+		}
+
+		gadget := &TopKubectlGadget[types.Stats]{
+			TopGadget: commontop.TopGadget[types.Stats]{
+				Name:           "biotop",
+				CommonTopFlags: &flags,
+				Parser:         parser,
+				ColMap:         cols.GetColumnMap(),
+			},
+			commonFlags: commonFlags,
+			nodeStats:   make(map[string][]*types.Stats),
+		}
+		gadget.Printer = gadget
```

Comment:
It is strange, but it seems the way to do it. Maybe we could move it with `TopLocalGadget.Run` and `TopKubectlGadget.Run` to avoid every gadget having to do this.